### PR TITLE
added encryption part

### PR DESCRIPTION
**Pull request by ashwinagalcha-ps**

### CODE REVIEW

File: src/main/scala/obfuscation/Encryption.scala [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+object Encryption {
+
+	private val KeySalt           = "3hoj3jon-jh31-o15c-3h61-8ars8dc1n3lw".getBytes(StandardCharsets.UTF_8)
+	private val AES               = "AES"
+	private val Algorithm         = AES + "/ECB/PKCS5Padding"
+	private val HashingIterations = 9999
+	private val KeySizeBits       = 128
+
+	def hashPassword(password: Array[Char],
```
Comment (AlMaior) [inline src/main/scala/obfuscation/Encryption.scala:39]: Add private

File: src/main/scala/obfuscation/Encryption.scala [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+			val keyFactory = SecretKeyFactory.getInstance(hashingAlgorithm)
+			val keySpec    = new PBEKeySpec(password, salt, iterations, keyLength)
+			val key        = keyFactory.generateSecret(keySpec)
+			key.getEncoded
+		} catch {
+			case e @ (_: NoSuchAlgorithmException | _: InvalidKeySpecException) => throw new RuntimeException("Password hashing error", e)
+		}
+	}
+
+	def prepareKey(password: String): SecretKeySpec = new SecretKeySpec(hashPassword(password.toCharArray, KeySalt), AES)
```
Comment (AlMaior) [inline src/main/scala/obfuscation/Encryption.scala:55]: Make it private

File: src/main/scala/obfuscation/Encryption.scala [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+	def encryptData(key: String, input: Any): String = {
```
Comment (AlMaior) [inline src/main/scala/obfuscation/Encryption.scala:57]: What about S3 solution which we talked about?

**Review by AlMaior — CHANGES_REQUESTED** [review]

File: src/main/scala/obfuscation/Encryption.scala (hunk 3 above)
Comment (ashwinagalcha-ps) [reply src/main/scala/obfuscation/Encryption.scala:57]: I think we should keep this flexible in terms of how users want to pass key. When we are encrypting/ decrypting one should read from S3 & pass key with required permissions.

**Review by AlMaior — APPROVED** [review]

**Pull request merged**